tests/discovery/Discovery.test.ts

```diff
@@ -59,7 +60,16 @@ describe('Discovery', () => {
   let nodeA: PolykeyAgent;
   let nodeB: PolykeyAgent;
   let identityId: IdentityId;
+
+  const mockedRefreshBucket = jest.spyOn(
+    NodeManager.prototype,
+    'refreshBucket',
+  );
```

Comment:
We want to stub it out. It was mostly to avoid extra logging. I'm not sure it's needed anymore but we do want to avoid a bunch of random connections in the background while testing discovery.

Reply:
So the discovery tests is making use of nodes domain which ultimately connects those nodes over the network. To do this, we have to simulate agents running. This is done in this test with `PolykeyAgent` class, and these agents are nodes that are running in-process. But they are bound to the OS network interfaces.

Reply:
Ok some notes about how we can eliminate these mocks in the PR comment, but it will require a new PR on changing some parts of the bootstrapping phase for nodes too.